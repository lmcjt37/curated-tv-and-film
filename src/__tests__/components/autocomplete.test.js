import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Autocomplete from '../../components/autocomplete';

const TestComponent = () => {
  const [autoComplete, setAutoComplete] = React.useState([]);
  const [, setShowDelete] = React.useState(false);

  const handleChange = (value) => {
    if (value) {
      setAutoComplete([{ label: 'Amazing!' }]);
    } else {
      setAutoComplete([]);
    }
  };
  return (
    <Autocomplete
      handleChange={handleChange}
      autoComplete={autoComplete}
      changeState={setShowDelete}
    />
  );
};

it('render all initial child components, where isOpen is false', async () => {
  render(<Autocomplete />);

  expect(screen.queryByTestId('suggestions')).toBeFalsy();
});

it('render all initial child components, where isOpen is true', async () => {
  render(<TestComponent />);

  await userEvent.type(screen.getByTestId('auto-complete-input'), 'a');

  await expect(screen.getByTestId('suggestions')).toBeTruthy();
  await expect(screen.getAllByText('Amazing!')).toBeTruthy();
});

//TODO - Tests functions

it('snapshot of initial component', () => {
  const { asFragment } = render(<Autocomplete />);
  expect(asFragment).toMatchSnapshot();
});
