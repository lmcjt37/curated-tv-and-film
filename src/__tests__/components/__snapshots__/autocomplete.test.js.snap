// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of initial component 1`] = `
<div
  aria-expanded="false"
  aria-haspopup="listbox"
  aria-labelledby="downshift-simple-label"
  class="Autocomplete-container-1"
  role="combobox"
>
  <div
    class="MuiInputBase-root Autocomplete-inputRoot-3 MuiInputBase-fullWidth"
  >
    <input
      aria-autocomplete="list"
      aria-labelledby="downshift-simple-label"
      autocomplete="off"
      class="MuiInputBase-input Autocomplete-inputInput-4"
      id="downshift-simple-input"
      placeholder="Search..."
      spellcheck="false"
      type="text"
      value=""
    />
  </div>
  <div
    aria-labelledby="downshift-simple-label"
    id="downshift-simple-menu"
    role="listbox"
  />
</div>
`;
